nixos/tests/systemd-confinement.nix

```diff
@@ -48,16 +48,22 @@ import ./make-test-python.nix {
       { config.confinement.mode = "chroot-only";
         testScript = ''
           with subtest("chroot-only confinement"):
-              paths = machine.succeed('chroot-exec ls -1 / | paste -sd,').strip()
-              assert_eq(paths, "bin,nix,run")
+              paths = machine.succeed('chroot-exec ls -Am /').strip()
+              assert_eq(paths, "bin, dev, etc, nix, proc, root, run, sys, usr, var")
+              # systemd’s setup_namespace() calls base_filesystem_create()
+              # which creates those usual top level directories.
+              # Check that they're empty.
+              paths = machine.succeed('chroot-exec ls -Am /dev /etc /proc /root /sys /usr /var | paste -sd,').strip()
+              assert_eq(paths, "/dev:,,/etc:,,/proc:,,/root:,,/sys:,,/usr:,,/var:")
               uid = machine.succeed('chroot-exec id -u').strip()
               assert_eq(uid, "0")
               machine.succeed("chroot-exec chown 65534 /bin")
         '';
       }
       { testScript = ''
           with subtest("full confinement with APIVFS"):
-              machine.fail("chroot-exec ls -l /etc")
+              paths = machine.succeed('chroot-exec ls -Am /etc').strip()
+              assert_eq(paths, "")
```

Comment:
Again, even with APIVFS, `/etc` should *not* be there if you don't *explicitly* specify it, eg. via `BindPaths` or `BindReadOnlyPaths`.